File: Hazel-ScriptCore/Source/Hazel/Renderer/DebugRenderer.cs

namespace Hazel
{
	public static class DebugRenderer
	{

		public static void DrawLine(Vector3 p0, Vector3 p1, Vector4 color)
		{
			unsafe { InternalCalls.DebugRenderer_DrawLine(&p0, &p1, &color); }
		}

		public static void DrawQuadBillboard(Vector3 translation, Vector2 size, Vector4 color)
		{
			unsafe { InternalCalls.DebugRenderer_DrawQuadBillboard(&translation, &size, &color); }
		}

		public static float LineWidth
		{
			set { unsafe { InternalCalls.DebugRenderer_SetLineWidth(value); } }
		}

	}
}

File: Hazel/src/Hazel/Renderer/DebugRenderer.cpp
#include "hzpch.h"
#include "DebugRenderer.h"

namespace Hazel {

	void DebugRenderer::DrawLine(const glm::vec3& p0, const glm::vec3& p1, const glm::vec4& color)
	{
		m_RenderQueue.emplace_back([p0, p1, color](Ref<Renderer2D> renderer)
		{
			renderer->DrawLine(p0, p1, color);
		});
	}

	void DebugRenderer::DrawQuadBillboard(const glm::vec3& translation, const glm::vec2& size, const glm::vec4& color)
	{
		m_RenderQueue.emplace_back([translation, size, color](Ref<Renderer2D> renderer)
		{
			renderer->DrawQuadBillboard(translation, size, color);
		});
	}

	void DebugRenderer::SetLineWidth(float thickness)
	{
		m_RenderQueue.emplace_back([thickness](Ref<Renderer2D> renderer)
		{
			renderer->SetLineWidth(thickness);
		});
	}

}
